packaging/linux/systemd/keybase.gui.service

```diff
@@ -33,6 +33,10 @@ ExecStartPost=/usr/bin/env systemctl --user unset-environment KEYBASE_AUTOSTART
 
 # We don't set Restart=, since a GUI crash is probably visible to the user.
 
+# Only kill main Electron process, which kills the others. But if that fails,
+# systemd will send a SIGKILL to all Electron processes.
+KillMode=process
```

Comment:
systemd was sending a SIGTERM to every electron process at the same time, and it did not like that, occasionally crashing (and since we don't set restart=always for gui, it didn't restart successfully). This change makes it so it only SIGTERMs the main process and then SIGKILLs everything later if it doesn't work.

Reply:
This still does not explain why KBFS was not starting up but that seems better with this change somehow. Will do some more testing.

Reply:
Why don't we set `restart=always`?

Reply:
Jack's decision, "since a GUI crash is probably visible to the user."

I think I agree as well, not 100% certain

Reply:
cc @eli-schwartz who may be interested, I think arch community uses the same unit files

Reply:
GUI crashes will at least mean the icon in the corner disappears, yes?

Restart=always is completely wrong anyhow, it means if you click "Quit", then because the program exited successfully but it wasn't `systemctl stop` which stopped it, it gets restarted. Restart=on-failure would cause it to restart on crashes and other abnormal exits, if that is something considered to be desirable.